lib/overcommit/hook/pre_commit/w3c_css_validator.rb

```diff
@@ -0,0 +1,74 @@
+module Overcommit::Hook::PreCommit
+  # Runs `w3c_validators` against any modified CSS files.
+  class W3cCssValidator < Base
+    def run
+      begin
+        require 'w3c_validators'
+      rescue LoadError
+        return :warn, 'w3c_validators not installed -- run `gem install w3c_validators`'
+      end
+
+      result_messages =
+        begin
+          applicable_files.collect do |path|
+            results = validator.validate_file(path)
+            messages = results.errors + results.warnings
+            messages.collect do |msg|
+              # Some warnings are not per-line, so use 0 as a default
+              line = Integer(msg.line || 0)
+
+              # Build message by hand to reduce noise from the validator response
+              text = "#{msg.type.to_s.upcase}; URI: #{path}; line #{line}: #{msg.message.strip}"
+              Overcommit::Hook::Message.new(msg.type, path, line, text)
+            end
+          end
+        rescue W3CValidators::ValidatorUnavailable => e
+          return :fail, e.message
+        rescue W3CValidators::ParsingError => e
+          return :fail, e.message
+        end
+
+      result_messages.flatten!
+      return :pass if result_messages.empty?
```

Comment:
Not necessarily:
```ruby
def process_hook_return_value(hook_return_value)
  if hook_return_value.is_a?(Array) &&
     hook_return_value.first.is_a?(Message)
    # Process messages into a status and output
    Overcommit::MessageProcessor.new(
      self,
      @config['problem_on_unmodified_line'],
    ).hook_result(hook_return_value)
  else
    # Otherwise return as-is
    hook_return_value
  end
end
```
This only uses `Overcommit::MessageProcessor` if `hook_return_value` is a _non-empty_ `Array` of `Overcommit::Hook::Message` objects. If it's empty, it's returned as-is (i.e. `[]` rather than `[:pass]`).